tasks/Resources/SERH_OLM/install.resource

```diff
@@ -0,0 +1,145 @@
+*** Settings ***
+Documentation    Collections of keywords to install/uninstall Managed Starburst
+Library          OpenShiftLibrary
+Resource         ../../../tests/Resources/Common.robot
+
+
+*** Variables ***
+${STARBURST_CR_DEFAULT_NAMESPACE}=    redhat-starburst-operator
+
+
+*** Keywords ***
+Check Managed Staburst Is Not Installed
+    [Documentation]    Checks Managed Staburst is Not installed on the cluster.
+    ...                Fails if it is
+    ${installed}=    Is Managed Starburst Installed
+    IF    ${installed} == ${TRUE}
+        Fail    msg=Managed Starburst is still installed or not fully Uninstalled
+    END
+
+Is Managed Starburst Installed
+    [Documentation]    Checks Managed Staburst is installed on the cluster.
+    ...                Returns TRUE if the resources created by the install process
+    ...                are found in the cluster
+    ${result}=  Run Keyword And Return Status
+    ...  Run Keywords
+    ...  Oc Get  kind=Project  field_selector=metadata.name=${STARBURST_CR_DEFAULT_NAMESPACE}  AND
+    ...  Oc Get  kind=CatalogSource  namespace=${STARBURST_CR_DEFAULT_NAMESPACE}
+    ...    field_selector=metadata.name=addon-managed-starburst-catalog
+    [Return]  ${result}
+
+Get Managed Starburst CR Status
+    [Documentation]     Get the installation stage by reading the "RHOAM installation object" YAML
+    ${status}=     Run    oc get starburstenterprise -n ${STARBURST_CR_DEFAULT_NAMESPACE} -o json | jq '.items[].status.conditions[] | .reason,.type' | sed 's/"//g'   # robocop: disable
+    ${status_list}=    Text To List    ${status}
+    # Run Keyword And Continue On Failure    List Should Contain Value    ${status_list}    Initialized
+    # Run Keyword And Continue On Failure    List Should Contain Value    ${status_list}    InstallSuccessful
+    # Run Keyword And Continue On Failure    List Should Contain Value    ${status_list}    Deployed
+    [Return]    ${status_list[-1]}
+
+Wait Until Managed Starburst CR Is Deployed
+    [Documentation]    Keeps checking for Starburst CustomResource status until
+    ...                it gets "Deployed"
+    [Arguments]    ${retries}=10   ${retries_interval}=30s
+    FOR  ${retry_idx}  IN RANGE  0  1+${retries}
+        Log    Checking Managed Starburst status: retry ${retry_idx}    console=yes
+        ${current_status}=   Get Managed Starburst CR Status
+        Log    Last status of Managed Starburst CR installation: ${current_status}    console=yes
+        ${completed_flag}=     Run Keyword And Return Status    Should Be Equal    ${current_status}    Deployed
+        Exit For Loop If    $completed_flag == True
+        Sleep    ${retries_interval}
+    END
+    IF    $completed_flag == False
+        Fail    Managed Starburst CR didn't get deployed after ${retries} retries.
+        ...     (time between retries: ${retries_interval}). Check the cluster...
+    END
```

Comment:
Wouldn't `Wait Until Keyword Succeeds    ${retries}x    ${retries_interval}    Get Managed Starburst CR Status` work as well?
Perhaps by slightly changing `Get Managed Starburst CR Status` to return pass/fail based on the internal status

Reply:
okay so by changing `Get Managed Starburst CR Status` into `Managed Starburst CR Status should be` and use it in combination with `Wait Until Keyword Succeeds`. Can do that